packages/transaction-controller/src/TransactionControllerIntegration.test.ts

```diff
@@ -0,0 +1,3369 @@
+      mockNetwork({
+        networkClientConfiguration: mainnetNetworkClientConfiguration,
+        mocks: [
+          // NetworkController
+          // BlockTracker
+          {
+            request: {
+              method: 'eth_blockNumber',
+              params: [],
+            },
+            response: {
+              result: '0x1',
+            },
+          },
+          // BlockTracker
+          {
+            request: {
+              method: 'eth_blockNumber',
+              params: [],
+            },
+            response: {
+              result: '0x2',
+            },
+          },
+        ],
+      });
+      const { networkController, transactionController } =
+        await newController();
+      const startTrackinSpy = jest.spyOn(
+        transactionController,
+        'startTrackingByNetworkClientId',
+      );
+
+      await networkController.upsertNetworkConfiguration(
+        {
+          ...networkClientConfiguration,
+          rpcUrl: 'https://mock.rpc.url',
+        },
+        { setActive: false, referrer: 'https://mock.referrer', source: 'dapp' },
+      );
+
+      expect(startTrackinSpy).toHaveBeenCalledTimes(1);
+      expect(transactionController).toBeDefined();
+    });
+    it('should stop tracking when a network is removed', async () => {
+      mockNetwork({
+        networkClientConfiguration: mainnetNetworkClientConfiguration,
+        mocks: [
+          // NetworkController
+          // BlockTracker
+          {
+            request: {
+              method: 'eth_blockNumber',
+              params: [],
+            },
+            response: {
+              result: '0x1',
+            },
+          },
+          // BlockTracker
+          {
+            request: {
+              method: 'eth_blockNumber',
+              params: [],
+            },
+            response: {
+              result: '0x2',
+            },
+          },
+        ],
+      });
+      const { networkController, transactionController } =
+        await newController();
+      const stopTrackinSpy = jest.spyOn(
+        transactionController,
+        'stopTrackingByNetworkClientId',
+      );
+
+      const configurationId =
+        await networkController.upsertNetworkConfiguration(
+          {
+            ...networkClientConfiguration,
+            rpcUrl: 'https://mock.rpc.url',
+          },
+          {
+            setActive: false,
+            referrer: 'https://mock.referrer',
+            source: 'dapp',
+          },
+        );
+
+      networkController.removeNetworkConfiguration(configurationId);
+
+      // advance time to trigger events
+      await advanceTime({ clock, duration: 1000 });
+
+      expect(stopTrackinSpy).toHaveBeenCalledTimes(1);
+      expect(transactionController).toBeDefined();
+    });
+  });
+
+  describe('startIncomingTransactionPolling', () => {
+    // TODO(JL): IncomingTransactionHelper doesn't populate networkClientId on the generated tx object. Should it?..
```

Comment:
actually this may be fine. Only transactions that don't already exist in state will be missing a networkClientId. Existing transactions, which would have a networkClientId, will be updated correctly with the new data from Etherscan